designs/aws-launch-template-network-interfaces.md

```diff
@@ -0,0 +1,126 @@
+      deleteOnTermination: false          # optional, Indicates whether the network interface is deleted when the instance is terminated.
+      description: "..."                  # optional, A description for the network interface.
+      deviceIndex: 0                      # required, The device index for the network interface attachment.
+      interfaceType: "interface"          # optional, The interface type for the network interface. To create an Elastic Fabric Adapter (EFA), specify efa
+      networkCardIndex: 0 
+      ipv4PrefixCount: 1                  # optional, The number of IPv4 prefixes that AWS automatically assigns to the network interface.
+      ipv6PrefixCount: 1                  # optional, The number of IPv6 prefixes that AWS automatically assigns to the network interface.
+    - { ... }
+```
+
+subnet and securityGroup will remain part of the `AWSNodeTemplate` and will be used for both network interfaces.
+in following iterations, as the `AWSNodeTemplate` move from `v1alpha1` to higher versions, we can add support for:
+
+```yaml
+subnetSelector: String
+securityGroupSelector: String
+```
+
+this way we can support different subnets and security groups for each network interface, we can also remove the limit
+of having only 2 network interfaces per LaunchTemplate.
+
+The [drift design doc #366](https://github.com/aws/karpenter-core/pull/366) suggests classifying CRD fields as  (1) Static, (2) Dynamic, or (3) Behavioral. in this
+context, the networkInterfaces would be classified as following:
+
+|                                                                | Static | Dynamic | Behavioral |
+|----------------------------------------------------------------|--------|---------|------------|
+| AWSNodeTemplate.NetworkInterfaces.length                       | x      |         |            |
+| AWSNodeTemplate.NetworkInterfaces.order                        | x      |         |            |
+| AWSNodeTemplate.NetworkInterfaces[@].AssociateCarrierIpAddress | x      |         |            |
+| AWSNodeTemplate.NetworkInterfaces[@].AssociatePublicIpAddress  | x      |         |            |
+| AWSNodeTemplate.NetworkInterfaces[@].DeleteOnTermination       |        |         | x          |
+| AWSNodeTemplate.NetworkInterfaces[@].Description               | x      |         |            |
+| AWSNodeTemplate.NetworkInterfaces[@].DeviceIndex               | x      |         |            |
+| AWSNodeTemplate.NetworkInterfaces[@].InterfaceType             | x      |         |            |
+| AWSNodeTemplate.NetworkInterfaces[@].NetworkCardIndex          | x      |         |            |
+| AWSNodeTemplate.NetworkInterfaces[@].ipv4PrefixCount           | x      |         |            |
+| AWSNodeTemplate.NetworkInterfaces[@].ipv6PrefixCount           | x      |         |            |
+
+since `AWSNodeTemplate.NetworkInterfaces` is a list, Drift would be triggered if:
```

Comment:
> The initial design is to default the behavior to terminate and replace. Maybe this is a reasonable case to consider in-place as it would be its own subspec? Would terminate and replace be too unreasonable as a initial landing place for this design?

Agree, it is a reasonable case for the in-place update, but it was mentioned by the original design doc for drift as a no-go. I think it is still reasonable for terminate and replace, except for DeleteOnTermination ([reason](https://github.com/aws/karpenter/pull/3819#discussion_r1229400842)) which is marked as `behavioral`